exchange/utils_test.go

```diff
@@ -1,11 +1,123 @@
 package exchange
 
 import (
+	"context"
+	"encoding/json"
 	"testing"
 
+	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbsmetrics"
+	"github.com/stretchr/testify/assert"
 )
 
+// permissionsMock mocks the Permissions interface for tests
+//
+// It always disallows BrightRoll for GDPR consent
+type permissionsMock struct{}
+
+func (p *permissionsMock) HostCookiesAllowed(ctx context.Context, consent string) (bool, error) {
+	return true, nil
+}
+
+func (p *permissionsMock) BidderSyncAllowed(ctx context.Context, bidder openrtb_ext.BidderName, consent string) (bool, error) {
+	return true, nil
+}
+
+func (p *permissionsMock) PersonalInfoAllowed(ctx context.Context, bidder openrtb_ext.BidderName, consent string) (bool, error) {
+	if bidder == "brightroll" {
```

Comment:
I would have it return true for "appnexus", and false otherwise. Otherwise a test that typos on "brightroll" will have strange behavior, and the real test returns false for everything but the specific bidders that were consented to.

Also help catch an issue if `corebidder` doesn't return the correct value here for some reason. As long as corebidder returns something other than "brightroll", we pass. We would rather tighten up to have corebidder return "appnexus", or else we fail.

Reply:
PR Updated!